easybuild/easyblocks/x/xalt.py

```diff
@@ -0,0 +1,92 @@
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for XALT, implemented as an easyblock
+@author: Scott McMillan (NVIDIA)
+"""
+import os
+import re
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM, MANDATORY
+from easybuild.tools.build_log import EasyBuildError
+
+
+class EB_XALT(ConfigureMake):
+    """Support for building and installing XALT."""
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            'config_py': [None, "XALT site filter file", MANDATORY],
+            'syshost': ['hardcode:cluster', "System name", CUSTOM],
+            'transmission': ['syslog', "", CUSTOM],
+        }
+        return ConfigureMake.extra_options(extra_vars)
+
+    def configure_step(self):
+        """Custom configuration step for XALT."""
+
+        # Remove the subdir from the install prefix to resolve the
+        # following configure error.  XALT automatically creates a
+        # versioned directory hierarchy.
+        #
+        #    Do not install XALT with 2.7.27 as part of the
+        #    prefix.  It will lead to many many problems with future
+        #    installs of XALT
+        #
+        #    Executables built with the current version of XALT
+        #    will not work with future installs of XALT!!!
+        #
+        #    If you feel that you know better than the developer
+        #    of XALT then you can configure XALT with the configure
+        #    --with-IrefuseToInstallXALTCorrectly=yes and set the prefix
+        #    to include the version
+        self.installdir = os.path.normpath(re.sub(self.install_subdir, '', self.installdir))
+
+        # XALT site filter config file is mandatory
+        if not self.cfg['config_py']:
+            raise EasyBuildError('XALT site filter config must be specified. '
+                                 'Use "--try-amend=config_py=<file>"')
+        self.cfg.update('configopts', '--with-config=%s' % self.cfg['config_py'])
```

Comment:
Throwing an error here makes sense, since there's no way we can come up with a generic configuration file.

I would change a couple of things here though:

* use positive logic
* check whether specified config.py actually exists
* improve the error message
   ```python
   config_py = self.cfg['config_py']
   if config_py:
        if os.path.exists(config_py):
            self.cfg.update('configopts', '--with-config=%s' % config_py)
        else:
            raise EasyBuildError("Specified XALT configuration file %s does not exist!", config_py)
    else:
        error_msg = "Location of XALT configuration file must be specified via 'config_py' easyconfig parameter. "
        error_msg += "You can edit the easyconfig file, or use 'eb --try-amend=config_py=<path>'. "
        error_msg += "See https://xalt.readthedocs.io/en/latest/030_site_filtering.html for more information."
        raise EasyBuildError(error_msg)